src/psygnal/_group.py

```diff
@@ -40,95 +42,45 @@ class EmissionInfo(NamedTuple):
-
-    This is ok:
-
-    >>> class Events(SignalGroup, strict=True):
-    ...     sig1 = Signal(int)
-    ...     sig1 = Signal(int)
-
-    This will raise an exception
-
-    >>> class Events(SignalGroup, strict=True):
-    ...     sig1 = Signal(int)
-    ...     sig1 = Signal(str)  # not the same signature
-    """
-
-    _signals_: ClassVar[Mapping[str, Signal]]
-    _uniform: ClassVar[bool] = False
-
-    # this is only here to indicate to mypy that all attributes are SignalInstances.
-    def __getattr__(self, name: str) -> SignalInstance:
-        raise AttributeError(
-            f"SignalGroup {type(self).__name__!r} has no attribute {name!r}"
-        )
-
-    def __init_subclass__(cls, strict: bool = False) -> None:
-        """Finds all Signal instances on the class and add them to `cls._signals_`."""
-        cls._signals_ = {}
-        for k in dir(cls):
-            v = getattr(cls, k)
-            if isinstance(v, Signal):
-                cls._signals_[k] = v
-
-        cls._uniform = _is_uniform(cls._signals_.values())
-        if strict and not cls._uniform:
-            raise TypeError(
-                "All Signals in a strict SignalGroup must have the same signature"
-            )
+    ```python
+    from psygnal import Signal, SignalRelay, SignalGroup
 
-        return super().__init_subclass__()
+    class MySignals(SignalGroup):
+        all_signals: SignalRelay  # change the public name of the SignalRelay attribute
+        sig1 = Signal()
+        sig2 = Signal()
+    """
 
-    def __init__(self, instance: Any = None, name: str | None = None) -> None:
-        super().__init__(
-            signature=(EmissionInfo,),
-            instance=instance,
-            name=name or self.__class__.__name__,
-        )
+    def __init__(
+        self, signals: Mapping[str, SignalInstance], instance: Any = None
+    ) -> None:
+        super().__init__(signature=(EmissionInfo,), instance=instance)
+        self._signals = signals
         self._sig_was_blocked: dict[str, bool] = {}
-        for _, sig in self.signals.items():
-            sig.connect(self._slot_relay, check_nargs=False, check_types=False)
-
-    def __len__(self) -> int:
-        return len(self._slots)
-
-    @property
-    def signals(self) -> dict[str, SignalInstance]:
-        """Return {name -> SignalInstance} map of all signal instances in this group."""
-        return {n: getattr(self, n) for n in type(self)._signals_}
 
-    @classmethod
-    def is_uniform(cls) -> bool:
-        """Return true if all signals in the group have the same signature."""
-        return cls._uniform
+        # silence any warnings about failed weakrefs (will occur in compiled version)
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            for sig in signals.values():
+                sig.connect(
+                    self._slot_relay, check_nargs=False, check_types=False, unique=True
+                )
```

Comment:
Maybe add to `connect` parameter that will allow silence warning, or provide a private context manager for this. 

(If I correctly understand that compiled version will emit warning even with this operator).